### PR TITLE
Add (official?) I2P

### PR DESCRIPTION
**Pull request by joelmo**
i2pd should probably be renamed to purpleli2p later

### CODE REVIEW

**Comment by edwtjo** [comment]
Just note that the official names for the respective routers for the i2p and purplei2p projects are i2p and i2pd respectively. Using i2p as attribute is better since those looking for purple will most likely understand the difference. You're missing a maintainer field and perhaps you could supplement the PR with a nixos service. Also did you test this? Perhaps add an assert for i686 Linux since this build seems amd64 dependant.

**Comment by joelmo** [comment]
I made some fixes. A nixos service would be nice, I may provide one later. I tested it on dingo (a869), so it should work for others.

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -1655,6 +1655,8 @@ let
 
   httptunnel = callPackage ../tools/networking/httptunnel { };
 
+  i2p = callPackage ../tools/networking/i2pd {};
```
Comment (edwtjo) [inline pkgs/top-level/all-packages.nix]: i2p folder

Comment (joelmo) [inline pkgs/top-level/all-packages.nix]: thanks, fixed

**Comment by edwtjo** [comment]
Great, thanks!

**Pull request merged**